pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -167,12 +172,10 @@ func (r *DNSPolicyReconciler) buildClusterGateway(ctx context.Context, clusterNa
 	}
 
 	for i, addr := range gatewayAddresses {
-		addrType := *addr.Type
-		if addrType == gateway.MultiClusterHostnameAddressType {
-			addrType = gatewayv1beta1.HostnameAddressType
-		}
-		if addrType == gateway.MultiClusterIPAddressType {
-			addrType = gatewayv1beta1.IPAddressType
+		addrType, multicluster := utils.AddressTypeToSingleCluster(addr)
+
+		if !multicluster {
+			addrType = *addr.Type
 		}
```

Comment:
We don't need this check here. This should work with any gateway and we don't need to check other types of addresses only our own and convert them to the appropriate single cluster type.

Reply:
That is true. The `AddressTypeToSingleCluster` is already doing that check